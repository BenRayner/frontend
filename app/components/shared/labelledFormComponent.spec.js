/* global describe, it, expect */
import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';

import labelled from './labelledFormComponent';

describe('labelledFormComponent', () => {
  it('renders DOM element children as expected', () => {
    const LabelledFormComponent = labelled('input');

    const component = TestUtils.renderIntoDocument(
      <LabelledFormComponent label="Test Input Field" />
    );

    const node = ReactDOM.findDOMNode(component); // eslint-disable-line react/no-find-dom-node
    expect(node).toMatchSnapshot();
  });

  it('exposes a `focus` function', () => {
    const LabelledFormComponent = labelled('input');

    const component = TestUtils.renderIntoDocument(
      <LabelledFormComponent label="Test Input Field" />
    );

    expect(Object.getOwnPropertyDescriptor(LabelledFormComponent.prototype, 'focus')).toMatchSnapshot();
    expect(() => component.focus()).not.toThrow();
  });

  it('exposes a `value` getter', () => {
    const LabelledFormComponent = labelled('input');

    const component = TestUtils.renderIntoDocument(
      <LabelledFormComponent label="Test Input Field" defaultValue="meow" />
    );

    expect(Object.getOwnPropertyDescriptor(LabelledFormComponent.prototype, 'value')).toMatchSnapshot();
    expect(component.value).toBe("meow");
  });

  it('exposes specified proxy methods', () => {
    const LabelledFormComponent = labelled('input');

    LabelledFormComponent.proxyMethods = ['select'];

    const component = TestUtils.renderIntoDocument(
      <LabelledFormComponent label="Test Input Field" />
    );

    expect(component.select).toBeInstanceOf(Function);
    expect(Object.getOwnPropertyDescriptor(component, 'select')).toMatchSnapshot();
    expect(() => component.select()).not.toThrow();
  });
});
