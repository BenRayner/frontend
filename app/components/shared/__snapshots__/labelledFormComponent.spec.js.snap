// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`labelledFormComponent exposes a \`focus\` function 1`] = `
Object {
  "configurable": true,
  "enumerable": false,
  "value": [Function],
  "writable": true,
}
`;

exports[`labelledFormComponent exposes a \`value\` getter 1`] = `
Object {
  "configurable": true,
  "enumerable": false,
  "get": [Function],
  "set": undefined,
}
`;

exports[`labelledFormComponent exposes specified proxy methods 1`] = `
Object {
  "configurable": true,
  "enumerable": true,
  "value": [Function],
  "writable": true,
}
`;

exports[`labelledFormComponent renders DOM element children as expected 1`] = `
<div
  data-reactroot=""
  class="mb2"
>
  <label>
    <div
      class="bold mb1"
    >
      <!-- react-text: 4 -->
      Test Input Field
      <!-- /react-text -->
    </div>
    <input
      class="input"
    />
  </label>
  <!-- react-empty: 6 -->
  <!-- react-empty: 7 -->
</div>
`;

exports[`labelledFormComponent renders React component children as expected 1`] = `
<div
  data-reactroot=""
  class="mb2"
>
  <label>
    <div
      class="bold mb1"
    >
      <!-- react-text: 4 -->
      Test Input Field
      <!-- /react-text -->
    </div>
    <pre>
      {"this-should-be":"passed through","className":"input"}
    </pre>
  </label>
  <!-- react-empty: 6 -->
  <!-- react-empty: 7 -->
</div>
`;
